cmd/soroban-cli/src/rpc/mod.rs

```diff
@@ -188,4 +226,45 @@ impl Client {
             .request("getLedgerEntry", rpc_params![base64_key])
             .await?)
     }
+
+    pub async fn get_events(
+        &self,
+        start_ledger: u32,
+        end_ledger: u32,
+        event_type: Option<EventType>,
+        contract_ids: &[String],
+        topics: &[String],
+        limit: Option<usize>,
+    ) -> Result<Option<GetEventsResponse>, Error> {
+        let mut filters = serde_json::Map::new();
+        filters.insert(
+            "type".to_string(),
+            match event_type {
+                Some(t) => match t {
+                    EventType::All => "",
+                    EventType::Contract => "contract",
+                    EventType::System => "system",
+                },
+                None => "",
+            }
+            .into(),
+        );
```

Comment:
*style*
Inserting a blank `"type": ""` and depending on that not being an error seems a bit dependent on a server implementation quirk, so it would be better to exclude if we could.

There's a few other options for how we could write this here, which would avoid inserting the empty key.

The flattest is probably:
```rust
match event_type {
  Some(EventType::Contract) => filters.insert("type".to_string(), "contract".into()),
  Some(EventType::System) => filters.insert("type".to_string(), "system".into()),
  _ => {},
};
```
But using the `_` matcher means we lose the completeness checking. We could either add the `Some(All)`, and `None` cases, or use a few chaining operators for:
```rust
event_type.and_then(|t| match t {
  EventType::All => None,
  EventType::Contract => Some("contract"),
  EventType::System => Some("system"),
}).map(|t| filters.insert("type".to_string(), t.into()))
```
Not really an issue though, so seems fine as is, if you want tbh.

Reply:
Lol I didn't expect something like this to be so convoluted :joy: your second solution is quite neat and clever, but wow that's quite a Rust-ism. What do you think about something a little more "traditional"?

```rust
        if event_type.is_some() && event_type.unwrap() != EventType::All {
            filters.insert(
                "type".to_string(),
                match event_type.unwrap() {
                    EventType::Contract => "contract",
                    EventType::System => "system",
                    EventType::All => "", // never happens, see condition
                }
                .into(),
            );
        }
```

(I wish I could do `if let Some(t) = event_type && t != EventType::All` but rustc doesn't like that.)

Reply:
Ok I wrote this and it's kinda ugly - gonna stick with your second approach. Thanks for teaching me about `and_then`!

Reply:
fwiw, I couldn't remember if I wanted `map` or what the name of the method that did that was. So I went to https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then, and dug around to find `and_then`, with the right type signature.